browser/android/brave_sync_worker.cc

```diff
@@ -115,6 +115,13 @@ void BraveSyncWorker::RequestSync(JNIEnv* env) {
     sync_service_observer_.AddObservation(service);
   }
 
+  // Upstream shows the notification when passphrase is required but not set,
+  // see SyncErrorNotifier.computeGoalNotificationState .
+  // Brave always set the passphrase eventually, so we don't need this
+  // notification. The same for scanning QR code and using the codewords.
+  service->GetUserSettings()
+      ->MarkPassphrasePromptMutedForCurrentProductVersion();
```

Comment:
This writes a specific major version info into prefs. And we need to look out for this if we ever bump major version, otherwise `IsPassphrasePromptMutedForCurrentProductVersion()` would return false.

According to 
```
class SyncServiceObserver {
 public:
  // When one of the following events occurs, OnStateChanged() is called.
  // Observers should query the service to determine what happened.
  // - The sync service initialized successfully.
  // - The sync servers are unavailable at this time.
  // - Credentials are now in flight for authentication.
  // - The data type configuration has started or ended.
  // - Sync shut down.
  // - Sync errors (passphrase, auth, unrecoverable, actionable, etc.).
  // - Encryption changes.
  virtual void OnStateChanged(SyncService* sync) {}
```
It will be called not just for initialization. Have you tried something like setup sync and then bump major version then intentionally making sync server unavailable to see if notification shows up? Or just changing types to sync.

Reply:
@darkdh,

thanks, I missed this point when published the PR.

The check by your steps shows that there is no a further warning:
1. I modified `GetCurrentMajorProductVersion`/`sync_user_settings_impl.cc` to give 127, one above the current major version
2. I forced `syncStateChanged` by disabling the network - it was fired.
3. Notification wasn't shown because check at `SyncErrorNotifier.syncStateChanged`=>`SyncErrorNotifier.computeGoalNotificationState` code will not hit 
```
if (mSyncService.isPassphraseRequiredForPreferredDataTypes()
        && !mSyncService.isPassphrasePromptMutedForCurrentProductVersion()) {
    return NotificationState.REQUIRE_PASSPHRASE;
}
```
`mSyncService.isPassphraseRequiredForPreferredDataTypes()` => ... => `SyncUserSettingsImpl::IsPassphraseRequired()` returns false.

Also I tried to re-create the Sync chain.
Result was the same, just `prefs` was `PassphrasePromptMutedProductVersion` overwritten to 127.

So I think, for the current conditions PR does what is expected.